managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java

```diff
@@ -2871,4 +2872,22 @@ public void testOpEntryAdd_toString_doesNotThrowNPE(){
                 ", dataLength=" + dataLength +
                 '}';
     }
+
+    @Test
+    public void testLedgerReachMaximumRolloverTime() throws Exception {
+        ManagedLedgerConfig config = new ManagedLedgerConfig();
+        config.setMinimumRolloverTime(1, TimeUnit.MILLISECONDS);
+        config.setMaximumRolloverTime(1, TimeUnit.SECONDS);
+
+        ManagedLedger ml = factory.open("ledger-reach-maximum-rollover-time", config);
+        long firstLedgerId = ((PositionImpl) ml.addEntry("test".getBytes())).getLedgerId();
+
+        // the ledger rollover scheduled time is between 1000 and 1050 ms,
+        // wait 1100 ms, the ledger should be rolled over.
+        Awaitility.await()
+                .atMost(1100, TimeUnit.MILLISECONDS)
```

Comment:
I am not sure that on CI  we can rely on making the difference in the order of 100ms

it is possible that the machine is very slow and then we will see the test failing

Reply:
Yes, I'll increase the time order of magnitude, we could set the maximum rollover time to 10 seconds and wait 11 seconds? Maybe 1 second is more stable?

Reply:
let's try

@lhotari what do you think ?

Reply:
In general, I think it's better not to test accurate timings. The test code is cleaner when `.atMost` is omitted. The default value is 10 seconds and that should be suitable for most cases where the operation is expected to complete in less than 10 seconds.